src/highlight.cpp

```diff
@@ -775,6 +775,20 @@ static void color_string_internal(const wcstring &buffstr, highlight_spec_t base
     }
 }
 
+// For Issue#8877
+// Valid commands following redirection symbols at command line start are colored
+// as $fish_color_command instead of $fish_color_error 
+static bool symbol_text_before_command(const wcstring &buffer, size_t before_cmd_len) {
+    wcstring before_str = buffer.substr(0, before_cmd_len);
+
+    if (before_str.find(L'>') != wcstring::npos
+     || before_str.find(L'<') != wcstring::npos) {
```

Comment:
You're potentially scanning the string twice here. Better to use `std::any_of` or a simple for loop.

Reply:
> You're potentially scanning the string twice here. Better to use `std::any_of` or a simple for loop.

Thank you, mqudsi. I will commit next.